flax/nnx/nnx/bridge/wrappers.py

```diff
@@ -55,43 +58,58 @@ def _functional_constructor(*args: tp.Any, **kwargs: tp.Any) -> Functional[M]:
   return _functional_constructor
 
 
-class LinenWrapper(Module):
+def shaped_init(module: Module, *args, **kwargs):
+  """To trigger init of all `LinenToNNX` module variables and return a wholesome state."""
+  assert callable(module)
+  _ = module(*args, **kwargs)
+  return nnx.split(module)
+
+
+class LinenToNNX(Module):
   def __init__(
     self,
     module: linen.Module,
-    *args: tp.Any,
     rngs: tp.Optional[Rngs] = None,
-    **kwargs: tp.Any,
   ):
     self.module = module
+    self.rngs = rngs
+    self._set_initializing(True)
 
-    _rngs = (
-      {name: stream.key.raw_value for name, stream in rngs.items()}
-      if rngs
-      else {}
-    )
-    # rename default to params
-    if 'params' not in _rngs and 'default' in _rngs:
-      _rngs['params'] = _rngs['default']
-      del _rngs['default']
-
-    variables = module.init(_rngs, *args, **kwargs)
-
-    self.states = {
-      collection: variableslib.variable_type(collection)(value)
-      for collection, value in variables.items()
-    }
+  def _set_initializing(self, initializing: bool):
+    for _, value in graph.iter_graph(self):
+      if isinstance(value, Object):
+        value._object__state._initializing = initializing
 
   def __call__(
     self, *args: Any, rngs: tp.Optional[Rngs] = None, **kwargs: Any
   ) -> Any:
-    _rngs = (
-      {name: stream.key.value for name, stream in rngs.items()} if rngs else {}
-    )
 
-    variables = {
-      collection: value.value for collection, value in self.states.items()
-    }
+    # Shape-based lazy init of the flax variables
+    if not rngs:
+      rngs = self.rngs
+    if self._object__state.initializing:
+      _rngs = (
+        {name: stream.key.raw_value for name, stream in rngs.items()}
```

Comment:
We need to generate new keys so Linen Modules get new RNG state every time.
```suggestion
        {name: stream() for name, stream in rngs.items()}
```

Reply:
Done.